app/workers/subject_metadata_worker.rb

```diff
@@ -0,0 +1,24 @@
+class SubjectMetadataWorker
+  include Sidekiq::Worker
+
+  def perform(set_id)
+
+    smses = SetMemberSubject.where(subject_set_id: set_id)
+
+    update_sms_priority_sql = <<-SQL
+      UPDATE set_member_subjects
+      SET    priority = CAST(subjects.metadata->>'#priority' AS INTEGER)
+      FROM   subjects
+      WHERE  subjects.id = set_member_subjects.subject_id
+      AND    set_member_subjects.id IN (?)
```

Comment:
```suggestion
      AND    set_member_subjects.subject_set_id = ?
```

Reply:
I assume that this only deals with 1 subject_set_id at a time, if more you'll need the IN (id1,id2) syntax.